src/components/__test__/button.spec.ts

```diff
@@ -79,4 +79,20 @@ describe('button', () => {
     testButtonElement.dispatchEvent(new Event('mouseenter'));
     expect(mockMouseOverHandler).toHaveBeenCalledTimes(1);
   });
+
+  it('data-label attribute', () => {
```

Comment:
Small nit, perhaps add a quick check to show that testButton2 will also get the attribute once `updateLabel` is called on it, even though it didn't have a label to start out with